### DISCUSSION TITLE
Custom icon for marker

### DISCUSSION THREAD

**Discussion by hamzaakram25** [geoman-io/leaflet-geoman#1099] [Questions & Help]
Hi, hope you are well.
first of all thanks for this plugin
wonderful work... 👍 
i want some help can i add custom icon on every marker when i pm:create it will show popup with dropdown in which i will select icon and it will set the icon on that marker(not all markers).
can you please help me out..
thanks...

**Comment by Falke-Design** [comment]
You can keep the leaflet id as reference and get the marker in store function after selecting the icon in the dropdown
```
function changeIcon(id){
	var layer = map._layers[id];
	console.log(layer);
	//layer.setIcon(icon);
}

map.on('pm:create',(e)=>{
   e.layer.bindPopup("<button onclick='changeIcon(\""+e.layer._leaflet_id+"\")'>Save</button>");
});

**Reply by hamzaakram25** [reply]
can you please give some example with jsfiddle link???
bcz i am beginner and want to complete this task.
hope you will not mind...
thanks appreciated.. :)

**Reply by Falke-Design** [reply]
Look into: https://jsfiddle.net/9dthk64s/
